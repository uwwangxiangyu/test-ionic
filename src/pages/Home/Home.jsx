import { IonButton, IonContent, IonHeader, IonItem, IonLabel, IonPage, IonTitle, IonToolbar } from '@ionic/react'
import './Home.css'
import React from 'react'
import Header from '../../components/Header'

const Home = ({ history }) => {
  return (
    <IonPage>
      <Header title={'Home'}/>
      <IonContent fullscreen>
        <IonHeader collapse="condense">
          <IonToolbar>
            <IonTitle size="large">Blank</IonTitle>
          </IonToolbar>
        </IonHeader>
        <IonItem>
          <IonButton onClick={e => {
            e.preventDefault()
            history.push('/login')
          }}>
            <IonLabel>Login</IonLabel>
          </IonButton>
        </IonItem>
        <IonItem>
          <IonButton onClick={e => {
            e.preventDefault()
            history.push('/Register')
          }}>
            <IonLabel>Register</IonLabel>
          </IonButton>
        </IonItem>
        <IonItem>
          <IonButton onClick={e => {
            e.preventDefault()
            history.push('/debug')
          }}>
            <IonLabel>Debug</IonLabel>
          </IonButton>
        </IonItem>
      </IonContent>
    </IonPage>
  )
}

export default Home
