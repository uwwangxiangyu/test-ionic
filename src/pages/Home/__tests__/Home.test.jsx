import { render } from '@testing-library/react'
import React from 'react'
import Home from '../Home'

test('Render Home page', async () => {
  const { container } = render(<Home history={[]}/>)

  expect(container).toMatchSnapshot()
})