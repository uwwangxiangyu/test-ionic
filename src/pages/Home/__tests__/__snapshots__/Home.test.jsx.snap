// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home page 1`] = `
<div>
  <div
    class="ion-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-title>
          Home
        </ion-title>
        <ion-avatar
          slot="end"
        >
          <img
            alt="avatar"
            src="https://staging-dot-scraper-1ccc.ue.r.appspot.com/instagram/avatar/arabella.rollison"
            style="padding: 8px;"
          />
        </ion-avatar>
      </ion-toolbar>
    </ion-header>
    <ion-content
      fullscreen="true"
    >
      <ion-header
        collapse="condense"
      >
        <ion-toolbar>
          <ion-title
            size="large"
          >
            Blank
          </ion-title>
        </ion-toolbar>
      </ion-header>
      <ion-item>
        <ion-button>
          <ion-label>
            Login
          </ion-label>
        </ion-button>
      </ion-item>
      <ion-item>
        <ion-button>
          <ion-label>
            Register
          </ion-label>
        </ion-button>
      </ion-item>
      <ion-item>
        <ion-button>
          <ion-label>
            Debug
          </ion-label>
        </ion-button>
      </ion-item>
    </ion-content>
  </div>
</div>
`;
